src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1042,7 +1042,7 @@ private static object DeepCopierHelper(Type t, object original)
         /// </summary>
         /// <param name="t">The type.</param>
         /// <returns>true if <paramref name="t"/> is serializable, false otherwise.</returns>
-        internal static bool HasSerializer(Type t)
+        public static bool HasSerializer(Type t)
```

Comment:
Thanks for the clarification. A few comments:
> The method is synchronous and has no side-effects;

Well, especially since it "appears" harmless from the API perspective, I wouldn't want to give clients the impression it is safe to call it normally from grain code. It is indeed side-effects free in terms of mutating state, but not necessarily in terms of perf (although I don't know how harmful at this point).

> Serialize is already public and it involves locking (see call to GetSerializer),

I think you mean `SerializeInner`, which I believe it is public and not internal only because the generated code uses it. Nevertheless *logically* it should not be public. The serialization is invoked by Orleans in very specific threads, and not in the same thread as the user code runs in, so the contention is extremely low or non-existent. If users call it, there might be contention here, and *might* affect the entire runtime in unforeseeable ways (in addition to being just bad to have blocking code in user code itself).

Having said that, I still believe it might not be worth exposing this surface in our API (unless of course there's concrete scenarios that cannot be fulfilled without this, but that should be treated as a separate issue so that others can weigh in too).